### ISSUE TITLE
Device disconnects wifi after a while, and is unable to reconnect.

### ISSUE THREAD

**Issue by herozero777** [nerves-networking/vintage_net#262]
**Describe the bug**
Library: `vintage_net_wizard`. device: Raspberry pi 4 B .When I turn the device on after around 1 hour the the device disconnects from the wifi and is unable to reconnect. I tried connecting to wifi both by using the AP module of `vintage_net_wizard` and hard-coding the SSID & PSK in `config.exs`. Few lines from error logs are:
"wlan0: internet test failed (:emfile)" can be see on the screen. It also states "dropping net with 'nil' SSID.... " 
 
**Error Traces**
 ----------------------------- Error trace on my ssh connected terminal ------------------
09:32:44.004 [debug] wlan0: Internet test failed: (:emfile)
        
09:32:44.004 [info]  RouteManager: set_connection_status wlan0 -> :lan
        
09:32:44.008 [error] GenServer VintageNet.RouteManager terminating
** (stop) :emfile
    :erlang.open_port({:spawn_executable, '/sbin/ip'}, [:stderr_to_stdout, {:env, [{'PATH', '/usr/sbin:/usr/bin:/sbin:/bin'}]}, :use_stdio, :exit_status, :binary, :hide, {:args, ["route", "del", "192.168.8.0/22", "table", "main", "metric", "20", "dev", "wlan0", "scope", "link"]}])
    (elixir 1.11.2) lib/system.ex:822: System.cmd/3
    (vintage_net 0.9.2) lib/vintage_net/route/ip_route.ex:189: VintageNet.Route.IPRoute.ip_cmd/1
    (vintage_net 0.9.2) lib/vintage_net/route_manager.ex:244: VintageNet.RouteManager.handle_delete/1
    (elixir 1.11.2) lib/enum.ex:786: Enum."-each/2-lists^foreach/1-0-"/2
    (vintage_net 0.9.2) lib/vintage_net/route_manager.ex:219: VintageNet.RouteManager.update_route_tables/1
    (vintage_net 0.9.2) lib/vintage_net/route_manager.ex:159: VintageNet.RouteManager.handle_call/3
    (stdlib 3.13.2) gen_server.erl:706: :gen_server.try_handle_call/4
Last message (from #PID<0.2185.0>): {:set_connection_status, "wlan0", :lan}
State: %VintageNet.RouteManager.State{interfaces: %{"wlan0" => %VintageNet.Route.InterfaceInfo{default_gateway: {192, 168, 10, 1}, interface_type: :wifi, ip_subnets: [{{192, 168, 8, 253}, 22}], status: :internet, weight: 0}}, prioritization: [ethernet: :internet, wifi: :internet, mobile: :internet, _: :internet, ethernet: :lan, wifi: :lan, mobile: :lan, _: :lan], route_state: %{"wlan0" => 100}, routes: [{:rule, 100, {192, 168, 8, 253}}, {:local_route, "wlan0", {192, 168, 8, 253}, 22, 0, 100}, {:local_route, "wlan0", {192, 168, 8, 253}, 22, 20, :main}, {:default_route, "wlan0", {192, 168, 10, 1}, 0, 100}, {:default_route, "wlan0", {192, 168, 10, 1}, 20, :main}]}
Client #PID<0.2185.0> is alive

    (stdlib 3.13.2) gen.erl:208: :gen.do_call/4
    (elixir 1.11.2) lib/gen_server.ex:1024: GenServer.call/3
    (vintage_net 0.9.2) lib/vintage_net/interface/internet_connectivity_checker.ex:138: VintageNet.Interface.InternetConnectivityChecker.set_connectivity/2
    (vintage_net 0.9.2) lib/vintage_net/interface/internet_connectivity_checker.ex:132: VintageNet.Interface.InternetConnectivityChecker.check_connectivity/1
    (vintage_net 0.9.2) lib/vintage_net/interface/internet_connectivity_checker.ex:61: VintageNet.Interface.InternetConnectivityChecker.handle_info/2
    (stdlib 3.13.2) gen_server.erl:680: :gen_server.try_dispatch/4
    (stdlib 3.13.2) gen_server.erl:756: :gen_server.handle_msg/6
    (stdlib 3.13.2) proc_lib.erl:226: :proc_lib.init_p_do_apply/3

------------------------------ error trace images ( traces on monitor- device terminal ) -------------------
![image](https://user-images.githubusercontent.com/22791921/105344460-cafaf100-5c04-11eb-968c-f986b714f910.png)

** When I try to stop and restart application **
![image](https://user-images.githubusercontent.com/22791921/105341366-d2b89680-5c00-11eb-9fdf-8a9c4b83396d.png)

**Expected behavior**
Even if connection drops, it should be able to reconnect automatically.

**VintageNet Info**
Before error --------------- 
Interface wlan0
  Type: VintageNetWiFi
  Present: true
  State: :configured (0:13:50)
  Connection: :internet (0:13:11)
  Addresses: 192.168.8.253/22, fe80::dea6:32ff:fec6:18b4/64
  Configuration:
    %{
      ipv4: %{method: :dhcp},
      type: VintageNetWiFi,
      vintage_net_wifi: %{
        networks: [
          %{key_mgmt: :wpa_psk, psk: "....", ssid: "SEPL Mobile"},
          %{key_mgmt: :wpa_psk, psk: "....", ssid: nil}
        ]
      }
    }
After error -----------------
![image](https://user-images.githubusercontent.com/22791921/105333770-eb707e80-5bf7-11eb-97a7-baa0fa99e1d2.png)

**Environment**
==> core
|nerves_bootstrap| Environment Package List
  Pkg:         nerves_system_br
  Vsn:         1.13.7
  Type:        system_platform
  BuildRunner: {nil, []}

  Pkg:         nerves_toolchain_ctng
  Vsn:         1.7.2
  Type:        toolchain_platform
  BuildRunner: {nil, []}

  Pkg:         nerves_toolchain_aarch64_unknown_linux_gnu
  Vsn:         1.3.2
  Type:        toolchain
  BuildRunner: {Nerves.Artifact.BuildRunners.Local, []}

  Pkg:         nerves_system_rpi4
  Vsn:         1.13.5
  Type:        system
  BuildRunner: {Nerves.Artifact.BuildRunners.Local, []}

|nerves_bootstrap| Loadpaths Start
Nerves environment
  MIX_TARGET:   rpi4
  MIX_ENV:      dev

**Additional context**

**Comment by fhunleth** [comment]
The `:emfile` error should mean that too many files are open at the same time. Once too many files are open, lots of things start failing as you'd imagine. `vintage_net` might be a victim to some other piece of code or it's somehow keeping files open and it shouldn't be.

Can you provide any other clues on what's happening over the hour? Like is a NIF doing something and forgetting to close files? Or maybe you have a library that opens a lot of files and the default max files (I don't know what this is offhand) is too small?

**Comment by herozero777** [comment]
Peace, 
Thanks a lot @fhunleth. you were really on point. The issue lies with the Elixir Logger. For our Logging we are using `Logger_file_backend` library. When I add it's backend to Logger in runtime: `Logger.add_backend( {LoggerFileBackend, :test_log} )` The file IO process is opened (which I observed from Phoenix LiveDashboard ).
![image](https://user-images.githubusercontent.com/22791921/105496219-02d66700-5cdf-11eb-81b2-cff197660736.png)

But the command `Logger.remove_backend( {LoggerFileBackend, :test_log} )` terminates the child process and deletes it, but that doesn't (for reasons not known to me) close the File descriptor/ process. I checked the commands remove_backend uses and it uses the code - That does the job: handler would something like `{LoggerFileBackend, :test_log}`.
```
   case Supervisor.terminate_child(@name, handler) do
      :ok ->
        _ = Supervisor.delete_child(@name, handler)
        :ok

      {:error, _} = error ->
        error
    end
```

I think I need to post this issue in elixir/Logger.
